### PR TITLE
Feature/shared model proto query responses

### PR DESCRIPTION
**Pull request by luckychess**
## What is this pull request?
Proto container for QueryResponse; proto implementation for AccountAssetResponse
   
## Details/Features
- Add QueryResponse proto container
- Add proto implementation for AccountAsset
- Add proto implementation for AccountAssetResponse

### CODE REVIEW

File: shared_model/backend/protobuf/query_responses/proto_query_response.hpp [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+#include "responses.pb.h"
+#include "utils/lazy_initializer.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T, typename Archive>
+auto loadQueryResponse(Archive &&ar) {
+  int which =
+      ar.GetDescriptor()->FindFieldByNumber(ar.response_case())->index();
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::QueryResponse::QueryResponseVariantType>(
```
Comment (muratovv) [inline shared_model/backend/protobuf/query_responses/proto_query_response.hpp]: Need to add check of the inheritance here

File: shared_model/backend/protobuf/query_responses/proto_query_response.hpp [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+      template <typename T>
+      using Lazy = detail::LazyInitializer<T>;
+
+      using LazyVariantType = Lazy<QueryResponseVariantType>;
+
+     public:
+      /// type of proto variant
+      using ProtoQueryResponseVariantType =
+          boost::variant<w<AccountAssetResponse>>;
```
Comment (muratovv) [inline shared_model/backend/protobuf/query_responses/proto_query_response.hpp]: Better to use variadic template for all polymorphic wrappers

File: shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp [hunk 3]
```diff
@@ -20,7 +20,7 @@
 
 #include "interfaces/commands/add_asset_quantity.hpp"
 
-#include "backend/protobuf/common_objects/amount.hpp"
+#include "backend/protobuf/common_objects/proto_amount.hpp"
```
Comment (l4l) [inline shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: Why?

File: shared_model/backend/protobuf/query_responses/proto_query_response.hpp (hunk 1 above)
Comment (l4l) [reply shared_model/backend/protobuf/query_responses/proto_query_response.hpp]: What do you mean?

File: shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp (hunk 3 above)
Comment (kamilsa) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: We have no convension about naming. It causes such situations. We probably need to decide on it asap

File: shared_model/backend/protobuf/common_objects/proto_account_asset.hpp [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+namespace shared_model {
+  namespace proto {
+    class AccountAsset final : public interface::AccountAsset {
+     public:
+      template <typename AccountAssetType>
+      explicit AccountAsset(AccountAssetType &&accountAssetType)
+          : protoAccountAsset_(
+                std::forward<AccountAssetType>(accountAssetType)),
+            accountId_([this] { return protoAccountAsset_->account_id(); }),
```
Comment (kamilsa) [inline shared_model/backend/protobuf/common_objects/proto_account_asset.hpp]: Is there a need to write Lazy initializations for every field?

File: shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp (hunk 3 above)
Comment (l4l) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: Yeh, we are. But still there's should be some meaningful reason for previous name change. And personally I'd like to this fix being reverted

Comment (kamilsa) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: Probably in separate PR we should fix it

Comment (luckychess) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: It was just unification. All files in protobuf directory start their names with proto and this one was only an exception.

Comment (l4l) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: It's common_objects, they definetely has no connection to protobuf

Comment (luckychess) [reply shared_model/backend/protobuf/commands/proto_add_asset_quantity.hpp]: It's proto implementation of common objects.

File: shared_model/backend/protobuf/queries/proto_get_account_assets.hpp [hunk 5]
```diff
@@ -48,7 +45,7 @@ namespace shared_model {
           : GetAccountAssets(std::move(o.proto_)) {}
 
       const interface::types::AccountIdType &accountId() const override {
-        return account_id_->account_id();
+        return asset_id_->account_id();
```
Comment (kamilsa) [inline shared_model/backend/protobuf/queries/proto_get_account_assets.hpp]: Looks confusing. Maybe rename?

**Pull request merged**